api/src/main/java/ai/djl/ndarray/NDManager.java

```diff
@@ -719,6 +719,27 @@ default NDArray decode(InputStream is) throws IOException {
         return NDSerializer.decode(this, is);
     }
 
+    /**
+     * Decodes a numpy .npy {@link NDArray} through byte array.
+     *
+     * @param bytes byte array to load from
+     * @return {@link NDArray}
+     */
+    default NDArray decodeNumpy(byte[] bytes) {
```

Comment:
1. We should push auto detection logic from NDList to this level and not expose this API.
2. If we fix the bug in NDList, user can always use `NDList.decode()`

Reply:
I tried to add auto detection in `NDSerializer.decode()`, but I realized that the performance won't be good.
1. I added ByteBuffer version due the performance issue. It's about 2-3 times faster than InputStream version. I didn't add ByteBuffer version for numpy.
2. I think we should keep it simple and only dealing with numpy and other file format in NDList.
3. If there is need to save .npy file format, we can add `NDList.Encoding.NUMPY` format( .npz file is just a zip file, user can extract .npy from .npz file)